bridges/YoutubeBridge.php

```
@@ -102,9 +137,105 @@ private function ytBridgeAddItem($vid, $title, $author, $desc, $time){
 		$item['uri'] = self::URI . 'watch?v=' . $vid;
 		$thumbnailUri = str_replace('/www.', '/img.', self::URI) . 'vi/' . $vid . '/0.jpg';
 		$item['content'] = '<a href="' . $item['uri'] . '"><img src="' . $thumbnailUri . '" /></a><br />' . $desc;
+		$this->ytBridgeAttachVideo($vid, $item);
 		$this->items[] = $item;
 	}
 
+	private function ytBridgeAttachVideo($vid, &$item) {
+		$attach_format = $this->getInput('attach_format');
+
+		if(empty($attach_format)) {
+			return;
+		}
```

Better check if the `youtube-dl` exe exists here, and fail loudly otherwise. Since the parameters are fixed.

---

It’s not actually that simple to do a cross-platform check if an executable exists on the $PATH. I’d say simply log whatever error is returned by running youtube-dl and it becomes pretty clear if the error means youtube-dl isn’t installed or on the $PATH.